src/main/java/io/cdap/plugin/ariba/source/connector/AribaConnectorConfig.java

```diff
@@ -143,36 +141,25 @@ public void validateCredentials(FailureCollector failureCollector) {
   }
 
   public final void validateToken(FailureCollector collector) {
-    String tokenUrl = "https://api.au.cloud.ariba.com/v2/oauth/token";
-    HttpUrl.Builder builder = HttpUrl.parse(tokenUrl).newBuilder();
-    URL url = builder.build().url();
+    AribaServices aribaServices = new AribaServices(this);
     try {
-      if (httpAribaTokenCall(url).code() != HttpURLConnection.HTTP_OK) {
+      String accessToken = aribaServices.getAccessToken();
+      URL viewTemplatesURL = HttpUrl.parse(this.getBaseURL()).
+        newBuilder()
+        .addPathSegments(METADATA_PATH)
+        .addPathSegments(String.format(PATH_SEGMENT, this.getSystemType()))
+        .addQueryParameter(ResourceConstants.PRODUCT, ResourceConstants.ANALYTICS)
+        .addQueryParameter(ResourceConstants.REALM, this.getRealm()).build().url();
+      OkHttpClient enhancedOkHttpClient = new OkHttpClient();
+      Request req = aribaServices.buildDataRequest(viewTemplatesURL, accessToken);
+      Response response = enhancedOkHttpClient.newCall(req).execute();
+      if (response.code() != HttpURLConnection.HTTP_OK) {
         collector.addFailure("Credentials are incorrect", "Please check the credentials");
       }
-    } catch (IOException e) {
-      LOG.error(" Unable to validate token", e);
+    } catch (IOException | AribaException e) {
+      collector.addFailure("Credentials are incorrect", "Please check the credentials");
+      throw new ValidationException(collector.getValidationFailures());
```

Comment:
no need to throw exception, failure has been added to the collector.

Reply:
removed